src/pulse/types.rs

```diff
@@ -57,22 +57,14 @@ pub mod cb {
 /// For types we only have pointers to. Use structs with names so there is at
 /// least pointer type safety.
 pub mod opaque {
-    #[repr(C)]
-    pub struct pa_mainloop;
-    #[repr(C)]
-    pub struct pa_mainloop_api;
-    #[repr(C)]
-    pub struct pa_context;
-    #[repr(C)]
-    pub struct pa_spawn_api;
-    #[repr(C)]
-    pub struct pa_threaded_mainloop;
-    #[repr(C)]
-    pub struct pa_proplist;
-    #[repr(C)]
-    pub struct pa_stream;
-    #[repr(C)]
-    pub struct pa_operation;
+    pub enum pa_mainloop {}
```

Comment:
Did you switch these to enum because rust's compiler says something about empty structs?

It kind of seems dirty to me to write this as enum on the rust side just because we don't want to fill in all of the fields of the struct since in C they're definitely not enums, even if it works out functionally the same. Empty structs even seem encouraged by the rust book, though it could be outdated: https://doc.rust-lang.org/book/structs.html#unit-like-structs

Did you see other people changing this to enum? If it seems to be "the rust way" these days, then I'm okay with making them enums, but otherwise I'd rather see it stay as struct if there's some other way to get rid of the error message.

Also kind of curious about removing ``` #[repr(C)]```, is this no longer encouraged? (I guess it could have never been encouraged and I could have been doing it out of fear :p)